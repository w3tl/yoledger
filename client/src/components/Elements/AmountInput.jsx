import React from 'react';
import PropTypes from 'prop-types';
import NumberFormat from 'react-number-format';
import { Input } from 'semantic-ui-react';

function AmountInput({
  name, value, onChange, ...attrs
}) {
  return (
    <NumberFormat
      value={value}
      customInput={Input}
      thousandSeparator
      onValueChange={
        ({ value: newValue }) => onChange(
          { target: { value: newValue } },
          { name, value: newValue },
        )}
      name={name}
      {...attrs}
    />
  );
}

AmountInput.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  onChange: PropTypes.func,
};

AmountInput.defaultProps = {
  value: 0,
  onChange: () => {},
};

export default AmountInput;
