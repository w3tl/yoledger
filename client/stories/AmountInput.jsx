import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { AmountInput } from '../src/components';

storiesOf('AmountInput', module)
  .add('default', () => <AmountInput />)
  .add('with label', () => <AmountInput label="Amount" />)
  .add('with value', () => <AmountInput value={10} onChange={action('onChange')} />);
