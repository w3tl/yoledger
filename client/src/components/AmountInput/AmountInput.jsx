import React from 'react';
import PropTypes from 'prop-types';
import NumberFormat from 'react-number-format';
import { withLabel } from '../Elements/utils';

function AmountInput({ value, onChange }) {
  return (
    <NumberFormat
      value={value}
      thousandSeparator
      onValueChange={({ value: newValue }) => onChange({ target: { value: newValue } })}
    />);
}

AmountInput.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  onChange: PropTypes.func,
};

AmountInput.defaultProps = {
  value: 0,
  onChange: () => {},
};

export default withLabel(AmountInput);
