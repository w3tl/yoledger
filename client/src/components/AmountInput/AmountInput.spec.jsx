import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AmountInput from './AmountInput';

configure({ adapter: new Adapter() });

describe('AmountInput', () => {
  it('should render correctly', () => {
    const output = shallow(
      <AmountInput value={10} />,
    );
    expect(output).toMatchSnapshot();
  });

  it('should handle the input event', () => {
    const onChange = jest.fn();
    const output = shallow(
      <AmountInput onChange={onChange} />,
    );
    output.simulate('change', 10.1);
    expect(onChange).toHaveBeenCalledWith(10.1);
  });
});
