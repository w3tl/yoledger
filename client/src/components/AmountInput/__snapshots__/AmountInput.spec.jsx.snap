// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountInput should render correctly 1`] = `
<AmountInput
  id=""
  onChange={[Function]}
  value={10}
/>
`;
